{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "26a83726-3635-4cf3-8470-a739e9e5ecc8",
   "metadata": {},
   "source": [
    "# Getting GWAS summary files\n",
    "* https://www.ebi.ac.uk/gwas\n",
    "* ant its API: https://www.ebi.ac.uk/gwas/docs/api\n",
    "* and pandasgwas package https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-023-09340-2\n",
    "* "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "695411b2-186a-4a7f-8a65-48b048562af6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# pip install this if you did not yet.\n",
    "from pandasgwas import get_studies\n",
    "from pandasgwas.summary_statistics import search, browser, download, parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ba709ea0-4575-4592-860c-409ec773e840",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100% (1 of 1) |##########################| Elapsed Time: 0:00:00 Time:  0:00:00\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>initialSampleSize</th>\n",
       "      <th>gxe</th>\n",
       "      <th>gxg</th>\n",
       "      <th>snpCount</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>imputed</th>\n",
       "      <th>pooled</th>\n",
       "      <th>studyDesignComment</th>\n",
       "      <th>accessionId</th>\n",
       "      <th>fullPvalueSet</th>\n",
       "      <th>userRequested</th>\n",
       "      <th>platforms</th>\n",
       "      <th>ancestries</th>\n",
       "      <th>genotypingTechnologies</th>\n",
       "      <th>replicationSampleSize</th>\n",
       "      <th>diseaseTrait.trait</th>\n",
       "      <th>publicationInfo.pubmedId</th>\n",
       "      <th>publicationInfo.publicationDate</th>\n",
       "      <th>publicationInfo.publication</th>\n",
       "      <th>publicationInfo.title</th>\n",
       "      <th>publicationInfo.author.fullname</th>\n",
       "      <th>publicationInfo.author.orcid</th>\n",
       "      <th>cohort</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6,990 European ancestry Bipolar disorder cases...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1.253e+06</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>GCST001877</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>[]</td>\n",
       "      <td>[{'type': 'initial', 'numberOfIndividuals': 61...</td>\n",
       "      <td>[{'genotypingTechnology': 'Genome-wide genotyp...</td>\n",
       "      <td>NA</td>\n",
       "      <td>Autism spectrum disorder, attention deficit-hy...</td>\n",
       "      <td>23453885</td>\n",
       "      <td>2013-02-27</td>\n",
       "      <td>Lancet</td>\n",
       "      <td>Identification of risk loci with shared effect...</td>\n",
       "      <td>Smoller JW</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>561 European, African American, Asian and othe...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>1.776e+06</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>GCST002368</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'manufacturer': 'Illumina'}]</td>\n",
       "      <td>[{'type': 'initial', 'numberOfIndividuals': 50...</td>\n",
       "      <td>[{'genotypingTechnology': 'Genome-wide genotyp...</td>\n",
       "      <td>982 European and other ancestry mothers of aut...</td>\n",
       "      <td>Maternal genotype effects in autism spectrum d...</td>\n",
       "      <td>24574247</td>\n",
       "      <td>2014-02-25</td>\n",
       "      <td>Autism Res</td>\n",
       "      <td>Investigation of maternal genotype effects in ...</td>\n",
       "      <td>Yuan H</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>42 Korean ancestry male cases and their parent...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>3.311e+05</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>GCST000991</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'manufacturer': 'Affymetrix'}]</td>\n",
       "      <td>[{'type': 'initial', 'numberOfIndividuals': 12...</td>\n",
       "      <td>[{'genotypingTechnology': 'Genome-wide genotyp...</td>\n",
       "      <td>NA</td>\n",
       "      <td>Autism spectrum disorders (language delay)</td>\n",
       "      <td>21519539</td>\n",
       "      <td>2011-02-25</td>\n",
       "      <td>Psychiatry Investig</td>\n",
       "      <td>Genome-wide association scan of korean autism ...</td>\n",
       "      <td>Cho SC</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1633 European ancestry trios, 942 trios</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>2.018e+06</td>\n",
       "      <td>up to</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>GCST002639</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'manufacturer': 'Illumina'}]</td>\n",
       "      <td>[{'type': 'initial', 'numberOfIndividuals': 48...</td>\n",
       "      <td>[{'genotypingTechnology': 'Genome-wide genotyp...</td>\n",
       "      <td>NA</td>\n",
       "      <td>Autism spectrum disorder-related traits</td>\n",
       "      <td>25534755</td>\n",
       "      <td>2014-09-30</td>\n",
       "      <td>Biol Psychiatry</td>\n",
       "      <td>A genome-wide association study of autism usin...</td>\n",
       "      <td>Chaste P</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>315 Han Chinese ancestry cases, 1,115 Han Chin...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>5.462e+05</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>GCST003130</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>[{'manufacturer': 'Affymetrix'}]</td>\n",
       "      <td>[{'type': 'initial', 'numberOfIndividuals': 14...</td>\n",
       "      <td>[{'genotypingTechnology': 'Genome-wide genotyp...</td>\n",
       "      <td>282 Han Chinese ancestry cases, 480 Han Chines...</td>\n",
       "      <td>Autism spectrum disorder</td>\n",
       "      <td>26398136</td>\n",
       "      <td>2015-09-23</td>\n",
       "      <td>PLoS One</td>\n",
       "      <td>Genome-Wide Association Study for Autism Spect...</td>\n",
       "      <td>Kuo PH</td>\n",
       "      <td>0000-0003-0365-3587</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  initialSampleSize                   gxe     gxg   snpCount  qualifier  imputed pooled studyDesignComment accessionId  fullPvalueSet userRequested             platforms                                 ancestries                                   genotypingTechnologies                             replicationSampleSize                                diseaseTrait.trait                 publicationInfo.pubmedId publicationInfo.publicationDate publicationInfo.publication               publicationInfo.title                publicationInfo.author.fullname publicationInfo.author.orcid cohort\n",
       "0  6,990 European ancestry Bipolar disorder cases...  False  False  1.253e+06     None     True   False        None         GCST001877      False          False                                   []  [{'type': 'initial', 'numberOfIndividuals': 61...  [{'genotypingTechnology': 'Genome-wide genotyp...                                                 NA  Autism spectrum disorder, attention deficit-hy...         23453885                    2013-02-27                             Lancet     Identification of risk loci with shared effect...            Smoller JW                               None       NaN \n",
       "1  561 European, African American, Asian and othe...  False  False  1.776e+06     None     True   False        None         GCST002368      False          False       [{'manufacturer': 'Illumina'}]  [{'type': 'initial', 'numberOfIndividuals': 50...  [{'genotypingTechnology': 'Genome-wide genotyp...  982 European and other ancestry mothers of aut...  Maternal genotype effects in autism spectrum d...         24574247                    2014-02-25                         Autism Res     Investigation of maternal genotype effects in ...                Yuan H                               None       NaN \n",
       "2  42 Korean ancestry male cases and their parent...  False  False  3.311e+05     None    False   False        None         GCST000991      False          False     [{'manufacturer': 'Affymetrix'}]  [{'type': 'initial', 'numberOfIndividuals': 12...  [{'genotypingTechnology': 'Genome-wide genotyp...                                                 NA         Autism spectrum disorders (language delay)         21519539                    2011-02-25                Psychiatry Investig     Genome-wide association scan of korean autism ...                Cho SC                               None       NaN \n",
       "3            1633 European ancestry trios, 942 trios  False  False  2.018e+06    up to    False   False        None         GCST002639      False          False       [{'manufacturer': 'Illumina'}]  [{'type': 'initial', 'numberOfIndividuals': 48...  [{'genotypingTechnology': 'Genome-wide genotyp...                                                 NA            Autism spectrum disorder-related traits         25534755                    2014-09-30                    Biol Psychiatry     A genome-wide association study of autism usin...              Chaste P                               None       NaN \n",
       "4  315 Han Chinese ancestry cases, 1,115 Han Chin...  False  False  5.462e+05     None    False   False        None         GCST003130      False          False     [{'manufacturer': 'Affymetrix'}]  [{'type': 'initial', 'numberOfIndividuals': 14...  [{'genotypingTechnology': 'Genome-wide genotyp...  282 Han Chinese ancestry cases, 480 Han Chines...                           Autism spectrum disorder         26398136                    2015-09-23                           PLoS One     Genome-Wide Association Study for Autism Spect...                Kuo PH                0000-0003-0365-3587       NaN "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "studies = get_studies(efo_trait = 'autism spectrum disorder')\n",
    "studies.studies.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3250d9a4-3055-471e-bbce-de3d137ab272",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(studies.studies)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "0daaf79a-d6fd-41da-8488-277ae291e277",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016596/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016596-EFO_0003888.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016596/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016596-EFO_0003888.h.tsv.gz  33686288    GCST90016596     EFO_0003888"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016599/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016599-EFO_0004215.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016599</td>\n",
       "      <td>EFO_0004215</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016599/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016599-EFO_0004215.h.tsv.gz  33686288    GCST90016599     EFO_0004215"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016602/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016602-EFO_0003756.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016602</td>\n",
       "      <td>EFO_0003756</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016602/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016602-EFO_0003756.h.tsv.gz  33686288    GCST90016602     EFO_0003756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016603/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016603-EFO_0003756.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016603</td>\n",
       "      <td>EFO_0003756</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016603/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016603-EFO_0003756.h.tsv.gz  33686288    GCST90016603     EFO_0003756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016606/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016606-EFO_0003756.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016606</td>\n",
       "      <td>EFO_0003756</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016606/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016606-EFO_0003756.h.tsv.gz  33686288    GCST90016606     EFO_0003756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016614/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016614-EFO_0003756.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016614</td>\n",
       "      <td>EFO_0003756</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016614/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016614-EFO_0003756.h.tsv.gz  33686288    GCST90016614     EFO_0003756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016620/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016620-EFO_0003756.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016620</td>\n",
       "      <td>EFO_0003756</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016620/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016620-EFO_0003756.h.tsv.gz  33686288    GCST90016620     EFO_0003756"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for idx in range(len(studies.studies)):\n",
    "    search_DF = search(PubMed_id=studies.studies[\"publicationInfo.pubmedId\"].loc[idx],\n",
    "                       study_accession_id=studies.studies[\"accessionId\"].loc[idx])\n",
    "    # some studies do not have data on the FTP server, so list only those which have\n",
    "    if len(search_DF) > 0:\n",
    "        display(search_DF.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8ccd4d81-4ead-4a19-84f6-e1558fce4c1e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# select one study\n",
    "search_DF = search(PubMed_id='33686288', study_accession_id='GCST90016596')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "3cb2aca2-b6bb-46bc-8919-ec67b569d95e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# will display data tree in the browser\n",
    "browser(search_DF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a09bb483-2ddc-4bdf-b906-4c83f55875b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>raw</th>\n",
       "      <th>dir</th>\n",
       "      <th>file_name</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>./GCST90016001-GCST90017000/GCST90016596/harmo...</td>\n",
       "      <td>/pub/databases/gwas/summary_statistics/GCST900...</td>\n",
       "      <td>33686288-GCST90016596-EFO_0003888.h.tsv.gz</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         raw                                                dir                                          file_name                  PubMed_id study_accession_id EFO_trait_id\n",
       "0  ./GCST90016001-GCST90017000/GCST90016596/harmo...  /pub/databases/gwas/summary_statistics/GCST900...  33686288-GCST90016596-EFO_0003888.h.tsv.gz  33686288    GCST90016596     EFO_0003888"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "baa28f75-1daa-4da4-a829-a990fcc715e3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "33686288-GCST90016596-EFO_0003888.h.tsv.gz(124.55 MB) downloaded in C:\\Users\\David Palecek\\pandasgwas_home\n"
     ]
    }
   ],
   "source": [
    "# this will downloaad around 200 MB \n",
    "download(search_DF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "fec8ff35-106c-49b0-9f83-b4b15e1d2eb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# and here is the data, this cell takes a bit of time to run (DF has 6 million lines)\n",
    "df = parse(search_DF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "d2ba0bf5-ea31-4a30-95c4-81c111200bcc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: center;\">\n",
       "      <th></th>\n",
       "      <th>hm_variant_id</th>\n",
       "      <th>hm_other_allele</th>\n",
       "      <th>hm_effect_allele</th>\n",
       "      <th>hm_beta</th>\n",
       "      <th>hm_odds_ratio</th>\n",
       "      <th>hm_ci_lower</th>\n",
       "      <th>hm_ci_upper</th>\n",
       "      <th>hm_effect_allele_frequency</th>\n",
       "      <th>hm_code</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>p_value</th>\n",
       "      <th>odds_ratio</th>\n",
       "      <th>ci_lower</th>\n",
       "      <th>ci_upper</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>PubMed_id</th>\n",
       "      <th>study_accession_id</th>\n",
       "      <th>EFO_trait_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_817186_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>rs3094315</td>\n",
       "      <td>1</td>\n",
       "      <td>817186</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_817341_A_G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>rs3131972</td>\n",
       "      <td>1</td>\n",
       "      <td>817341</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.004</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_817514_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>rs3131971</td>\n",
       "      <td>1</td>\n",
       "      <td>817514</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_818045_T_C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11</td>\n",
       "      <td>rs3131970</td>\n",
       "      <td>1</td>\n",
       "      <td>818045</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_818161_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>rs2073813</td>\n",
       "      <td>1</td>\n",
       "      <td>818161</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5797742</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>rs76141896</td>\n",
       "      <td>22</td>\n",
       "      <td>50392358</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>-0.007</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5797743</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>rs2273254</td>\n",
       "      <td>22</td>\n",
       "      <td>50435117</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5797744</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>rs2273255</td>\n",
       "      <td>22</td>\n",
       "      <td>50435128</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5797745</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>rs2273256</td>\n",
       "      <td>22</td>\n",
       "      <td>50435132</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.36</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5797746</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>rs5770920</td>\n",
       "      <td>22</td>\n",
       "      <td>50440197</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.003</td>\n",
       "      <td>0.24</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33686288</td>\n",
       "      <td>GCST90016596</td>\n",
       "      <td>EFO_0003888</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5797747 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        hm_variant_id hm_other_allele hm_effect_allele  hm_beta  hm_odds_ratio  hm_ci_lower  hm_ci_upper  hm_effect_allele_frequency  hm_code  variant_id  chromosome  base_pair_location effect_allele other_allele  beta   standard_error  p_value  odds_ratio  ci_lower  ci_upper  effect_allele_frequency PubMed_id study_accession_id EFO_trait_id\n",
       "0        1_817186_G_A          G               A        -0.003       NaN           NaN          NaN                  NaN                11      rs3094315       1             817186             G            A       0.003       0.003       0.19       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "1        1_817341_A_G          A               G        -0.004       NaN           NaN          NaN                  NaN                11      rs3131972       1             817341             A            G       0.004       0.003       0.13       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "2        1_817514_T_C          T               C        -0.004       NaN           NaN          NaN                  NaN                11      rs3131971       1             817514             T            C       0.004       0.003       0.17       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "3        1_818045_T_C          T               C        -0.002       NaN           NaN          NaN                  NaN                11      rs3131970       1             818045             T            C       0.002       0.003       0.53       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "4        1_818161_G_A          G               A         0.002       NaN           NaN          NaN                  NaN                10      rs2073813       1             818161             A            G       0.002       0.003       0.40       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "...               ...             ...              ...      ...            ...          ...          ...                         ...      ...         ...         ...                 ...           ...          ...    ...             ...      ...         ...       ...       ...                      ...       ...                ...          ...\n",
       "5797742           NaN        NaN             NaN           NaN       NaN           NaN          NaN                  NaN                15     rs76141896      22           50392358             C            A      -0.007       0.003       0.01       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "5797743           NaN        NaN             NaN           NaN       NaN           NaN          NaN                  NaN                15      rs2273254      22           50435117             T            C       0.002       0.003       0.36       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "5797744           NaN        NaN             NaN           NaN       NaN           NaN          NaN                  NaN                15      rs2273255      22           50435128             G            A       0.002       0.003       0.36       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "5797745           NaN        NaN             NaN           NaN       NaN           NaN          NaN                  NaN                15      rs2273256      22           50435132             T            C       0.002       0.003       0.36       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "5797746           NaN        NaN             NaN           NaN       NaN           NaN          NaN                  NaN                15      rs5770920      22           50440197             C            A       0.003       0.003       0.24       NaN        NaN       NaN              NaN             33686288    GCST90016596     EFO_0003888\n",
       "\n",
       "[5797747 rows x 24 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0cac4d37-aab0-451e-9d2b-09ba342a42b5",
   "metadata": {},
   "source": [
    "## TODO\n",
    "* try to plot various plots from this dataframe\n",
    "* and find out what they mean.\n",
    "* seaborn is a module which focuses on plotting from pandas dataframes.\n",
    "* also check out pandas documentation itself.\n",
    "* https://pandas.pydata.org/docs/getting_started/index.html\n",
    "* first test your plots on smaller DF where you take just first 200 lines or something"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9cccbe8-2525-422d-8b37-97c76de8f305",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pyualg",
   "language": "python",
   "name": "pyualg"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
